### ISSUE TITLE
Configuration to control where inspec is allowed to write to disk

### ISSUE THREAD

**Issue by phiggins** [inspec/inspec#3455]
### 🎛 Description

**Choose one:** is this a 🐛 bug report or 🙋 feature request?

This is a feature request.

As I mentioned in #3263, I've recently been doing work to allow running inspec with lowered privileges. Running inspec as the linux standard `nobody` user, I've found that inspec expects to be able to write to various places on disk like the user's home directory and the system temp directory.

Inspec offers some settings like the `--vendor-cache` option and `INSPEC_CONFIG_DIR` env variable to control some file writing locations, but there is no unified way to do this. I ended up not using these and setting the HOME and TMPDIR env variables that inspec uses implicitly.

Ideally for my use case inspec would have a single env variable or command line option that controlled where inspec was allowed to write to disk. I mentioned this idea to @chris-rock and he agreed this was worth opening an issue for.

**Comment by frezbo** [comment]
@phiggins I really like your idea and ability to run inspec cli commands like `archive`, `json` but most compliance testing would need root/sudo access.

**Comment by chris-rock** [comment]
@frezbo You're right, for local execution you would need to have access. I think this improvement would help us to be more predictive. e.g. InSpec could be optimized for selinux or appamor.

**Comment by aaronlippold** [comment]
when we move inspec into production and operational deployments - it would be reasonable to expect that we will need to provide selinux and apparmor configuration for proper inspec functioning and more importantly provide a way for us to help maintain security and proper function at both elevated and non-elevated privileges. This will require us to make proper adjustments in the resources such that they correctly are able to report in these contexts which tests failed due to security labeling (selinux and apparmor ) as well as failing due to insufficient privilege.

In my opinion, this is a critical back end component that will affect not only the ongoing accreditation and validation of inspec as a tool but its perception of usefulness and trust. If resources are not able to clearly communicate to end users and operators that they did not 'break' but were unable to execute due to valid security setting on the system - the perception will quickly become 'InSpec is broken'.
